radixdlt-core/radixdlt/src/main/java/com/radixdlt/consensus/UnverifiedVertex.java

```diff
@@ -109,30 +111,36 @@
 	private final BFTNode proposer;
 
 	@JsonCreator
-	UnverifiedVertex(
-		@JsonProperty("qc") QuorumCertificate qc,
-		@JsonProperty("view") Long viewId,
+	private UnverifiedVertex(
+		@JsonProperty(value = "qc", required = true) QuorumCertificate qc,
+		@JsonProperty("view") long viewId,
 		@JsonProperty("txns") List<byte[]> txns,
 		@JsonProperty("p") byte[] proposer,
 		@JsonProperty("tout") Boolean proposerTimedOut
 	) throws PublicKeyException {
 		this(
 			qc,
-			viewId != null ? View.of(viewId) : null,
+			View.of(viewId),
 			txns == null ? List.of() : txns,
 			proposer != null ? BFTNode.fromPublicKeyBytes(proposer) : null,
 			proposerTimedOut
 		);
 	}
 
-	public UnverifiedVertex(QuorumCertificate qc, View view, List<byte[]> txns, BFTNode proposer, Boolean proposerTimedOut) {
-		this.qc = Objects.requireNonNull(qc);
-		this.view = Objects.requireNonNull(view);
+	private UnverifiedVertex(QuorumCertificate qc, View view, List<byte[]> txns, BFTNode proposer, Boolean proposerTimedOut) {
+		this.qc = requireNonNull(qc);
+		this.view = requireNonNull(view);
 
 		if (proposerTimedOut != null && proposerTimedOut && !txns.isEmpty()) {
 			throw new IllegalArgumentException("Txns must be empty if timeout");
 		}
 
+		if (txns != null && !txns.isEmpty()) {
+			for (var txn : txns) {
+				requireNonNull(txn);
```

Comment:
But we can't assume we will receive only message serialised but the system, can we?

Reply:
Ah I think you mean the deserialisation process won't create a List with null inside? But this isn't the only way of creating the object and allowing this could cause unexpected behaviour in the system.